yt_dlp/extractor/bitchute.py

```diff
@@ -100,51 +92,82 @@ def _real_extract(self, url):
+    _VALID_URL = r'https?://(?:www\.)?bitchute\.com/(?P<type>channel|playlist)/(?P<id>[^/?#&]+)'
+    _TESTS = [{
+        'url': 'https://www.bitchute.com/channel/bitchute/',
+        'playlist_mincount': 10,
         'info_dict': {
-            'id': 'victoriaxrave',
+            'id': 'bitchute',
         },
-    }
+    }, {
+        'url': 'https://www.bitchute.com/playlist/wV9Imujxasw9/',
+        'playlist_mincount': 30,
+        'info_dict': {
+            'id': 'wV9Imujxasw9',
+        }
+    }]
 
     _TOKEN = 'zyG6tQcGPE5swyAEFLqKUwMuMMuF6IO2DZ6ZDQjGfsL0e4dcTLwqkTTul05Jdve7'
+    HTML_CLASS_NAMES = {
+        'channel': {
+            'container': 'channel-videos-container',
+            'title': 'channel-videos-title',
+            'description': 'channel-videos-text',
+        },
+        'playlist': {
+            'container': 'playlist-video',
+            'title': 'title',
+            'description': 'description',
+        }
+
+    }
 
-    def _entries(self, channel_id):
-        channel_url = 'https://www.bitchute.com/channel/%s/' % channel_id
+    def _entries(self, playlist_type, playlist_id):
+        playlist_url = 'https://www.bitchute.com/%s/%s/' % (playlist_type, playlist_id)
         offset = 0
         for page_num in itertools.count(1):
             data = self._download_json(
-                '%sextend/' % channel_url, channel_id,
-                'Downloading channel page %d' % page_num,
+                '%sextend/' % playlist_url, playlist_id,
+                'Downloading %s page %d' % (playlist_type, page_num),
                 data=urlencode_postdata({
                     'csrfmiddlewaretoken': self._TOKEN,
                     'name': '',
                     'offset': offset,
                 }), headers={
                     'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
-                    'Referer': channel_url,
+                    'Referer': playlist_url,
                     'X-Requested-With': 'XMLHttpRequest',
                     'Cookie': 'csrftoken=%s' % self._TOKEN,
                 })
-            if data.get('success') is False:
+            if data.get('success') is not True:
                 break
             html = data.get('html')
-            if not html:
-                break
-            video_ids = re.findall(
-                r'class=["\']channel-videos-image-container[^>]+>\s*<a\b[^>]+\bhref=["\']/video/([^"\'/]+)',
-                html)
-            if not video_ids:
-                break
-            offset += len(video_ids)
-            for video_id in video_ids:
-                yield self.url_result(
+            results = []
+            class_name = self.HTML_CLASS_NAMES[playlist_type]
+            for video_html in get_elements_html_by_class(class_name['container'], html):
+                match = re.search(r'<a\b[^>]+\bhref=["\']/video/(?P<id>[^"\'/]+)', video_html)
+                if not match:
+                    continue
```

Comment:
Old code was just looking for the video ids. Since the html provides more metadata (title, description, duration, view count) it can be extracted as well. When using e.g. `--match-filter "title ~= <keyword>"`  yt-dlp does not need to extract t every single video in the playlist which saves time since bitchute requests are not the fastest.

Reply:
I was asking about the specific code that I highlighted in my comment. The old code was extracting it with regex, and the new code with get_elements_html_by_class, but ends up giving more complicated code. So I was wondering if there was a reason you changed it

Anyway, never mind. It doesnt really matter